core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -661,4 +677,99 @@ public boolean equals(Object obj) {
       }
    }
 
+   void assertLockStream(BiConsumer<Cache<Object, Object>, CacheEntry<Object, Object>> consumer) throws Throwable {
+      for (int i = 0; i < 10; i++) {
+         cache.put(i, "value" + i);
+      }
+
+      CyclicBarrier barrier = new CyclicBarrier(2);
+
+      int key = 4;
+
+      LockedStream<Object, Object> stream = cache.getAdvancedCache().lockedStream();
+      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
+         if (e.getKey().equals(key)) {
+            try {
+               barrier.await(10, TimeUnit.SECONDS);
```

Comment:
The test is doing exactly what I thought it would, testing that if a consumer is currently running that it would block another write operation.

I however didn't think to verify that after the inner consumer completed that the lock was still held. I can add a check for the lock manager to verify that though as an extra extra precaution.